deployments/dex.go

```diff
@@ -0,0 +1,289 @@
+package deployments
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/epinio/epinio/helpers"
+	"github.com/epinio/epinio/helpers/kubernetes"
+	"github.com/epinio/epinio/helpers/termui"
+	"github.com/epinio/epinio/internal/auth"
+	"github.com/epinio/epinio/internal/duration"
+	"github.com/kyokomi/emoji"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Dex struct {
+	Debug   bool
+	Timeout time.Duration
+}
+
+var _ kubernetes.Deployment = &Dex{}
+
+const (
+	DexDeploymentID = "dex"
+	dexVersion      = "2.30.0"
+	dexChartFile    = "dex-0.6.0.tgz"
+)
+
+func (k *Dex) ID() string {
+	return DexDeploymentID
+}
+
+func (k *Dex) Backup(ctx context.Context, c *kubernetes.Cluster, ui *termui.UI, d string) error {
```

Comment:
Backup/Restore are no longer part of the interface in main, instead PreInstallCheck is needed

Reply:
Will remove (I think you mean `PreDeployCheck`, which [is already there](https://github.com/epinio/epinio/pull/668/files#diff-e77a4e78f949937daf2ca120d63abb5af20e28cb27c1ad16427b3f8277b39667R255))